wgpu-core/src/resource.rs

```diff
@@ -1391,6 +1394,22 @@ impl Global {
             hal_command_encoder_callback(None)
         }
     }
+
+    /// # Safety
+    ///
+    /// - The raw queue handle must not be manually destroyed
+    pub unsafe fn queue_as_hal<A: HalApi, F: FnOnce(Option<&A::Queue>) -> R, R>(
```

Comment:
Nit: lets use a `where` here, as the expressions are complex.

Reply:
Thank you!

Reply:
Nit on top of nit: all other `as_hal()` functions have the bound directly on the type definition, it would be more consistent to keep it or replace it across the code-base atomically?